crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -209,6 +216,8 @@ pub enum NextState<S: FreelyMutableState> {
     Unchanged,
     /// There is a pending transition for state `S`
     Pending(S),
+    /// There is a pending removal of the [`State<S>`] resource
+    Remove,
```

Comment:
Reference how only some states support this option and note how others handle it